### ISSUE TITLE
Better LaTeX Environment Detection

### ISSUE THREAD

**Issue by zcysxy** [tth05/obsidian-completr#15]
Thanks for this plugin! I really rely on it to write math in Obsidian. I'm thinking is it possible to improve the detection of the LaTeX environment? Currently the LaTeX provider will be triggered after any dollar sign, even in **code blocks**. So I think the better logic is to ignore dollar signs in inline code, code fences, and escaped dollar signs `\$`.

**Comment by tth05** [comment]
Thanks, that's probably a good idea. I'll get back to working on this this weekend. If understand correctly, you're talking about `` ` `` and `` ``` ``, right?

**Comment by zcysxy** [comment]
> Thanks, that's probably a good idea. I'll get back to working on this this weekend. If understand correctly, you're talking about `` ` `` and `` ``` ``, right?

Yes! And also the escaped dollar signs `\$`. For example, the dollar sign in `It costs me 15\$.` should not trigger the LaTeX provider.

**Comment by tth05** [comment]
Implemented in 45e9bf8de49a0b20b60e9b2898a407a9a512efde. I made as an option because one might need this when writing admonition blocks or something similar. Let me know if it works how you'd expect to work.

**Comment by zcysxy** [comment]
Yeah it works as expected. Thanks a lot!

**Comment by fmbarina** [comment]
Hi, sorry if reviving an issue like this is considered inappropriate, I just wanted to add another possible enhancement.

LaTeX completion becomes enabled for the entire file if you use $$ (double dollar signs) in the document front matter, as a value for example. I don't think makes much sense? The rest of obsidian doesn't seem to think the entire document is LaTeX, so maybe Completr should follow that convention as well.

Idea: add another option to ignore LaTeX completion trigger if inside front matter

Q: That's a very weird case, how did stumble upon that?
A: I have a field called `cost` and wanted to use `$`, `$$`, or `$$$` to indicate cost, so this is probably a very rare and maybe even unimportant issue.

**Comment by tth05** [comment]
Very rare use-case indeed, but should be fixed now!

**Comment by fmbarina** [comment]
Hi, that is some fast development!

Thank you for taking such a rare case so seriously. I'm afraid LaTeX completion is completely broken for me now? Coding is Fun. Maybe it's something I did, but after updating, I can't trigger LaTeX completion. To be clear, this also happens without the front matter block. I checked the manifest, and it says `"version": "2.8.1"`. I also checked the settings, and completion is on, just as it was before.

(no "begin" suggested)
![image](https://user-images.githubusercontent.com/70731450/173096601-8c37167f-7a26-4521-bd52-cd2542935209.png)

If you think this should be a separate issue or that there's anything I can do to help, just say the word.
